opengever/maintenance/scripts/fix_non_persisted_values.py

```diff
@@ -0,0 +1,955 @@
+        have been the current date at that time.
+        """
+        assert obj.portal_type in ('opengever.dossier.dossiertemplate', )
+        created_date = obj.created().asdatetime().date()
+
+        # Verify that the value is valid
+        bound_field = field.bind(obj)
+        bound_field.validate(created_date)
+
+        return created_date
+
+    def get_dossier_manager_value(self, obj, field):
+        if obj.reading == [] and obj.reading_and_writing == []:
+            return None
+
+        print "Encountered a problematic dossier_manager field:"
+        print "The object at %s doesn't have a value persisted" % obj
+        print "for its dossier_manager field, but at the same time"
+        print "has non-empty values for either the reading or "
+        print "reading_and_writing field."
+        print "This is a situation that can't be fixed automatically and"
+        print "needs to be looked at and corrected manually"
+
+        raise Exception("Unexpected dossier_manager settings")
+
+    # For these four acquired restricted defaults, we simply persist the
+    # volatile value. Because of the way these acquired defaults work, this
+    # might not necessarily be what the user saw when they first saved the
+    # form. But it's what last got displayed and effectively been used. It's
+    # the best we can do.
+
+    def get_classification_value(self, obj, field):
+        """Get value for classification field.
+        """
+        return self._get_volatile_value(obj, field)
+
+    def get_custody_period_value(self, obj, field):
+        """Get value for custody_period field.
+        """
+        return self._get_volatile_value(obj, field)
+
+    def get_retention_period_value(self, obj, field):
+        """Get value for retention_period field.
+        """
+        return self._get_volatile_value(obj, field)
+
+    def get_privacy_layer_value(self, obj, field):
+        """Get value for privacy_layer field.
+        """
+        return self._get_volatile_value(obj, field)
+
+    handlers = {
+        ('IDocumentMetadata', 'preserved_as_paper'): get_preserved_as_paper_value,  # noqa
+        ('IDocumentMetadata', 'document_date'): get_document_date_value,
+        ('start'): get_dossier_start_value,
+        ('IProtectDossier', 'dossier_manager'): get_dossier_manager_value,
+        ('ITask', 'deadline'): get_deadline_value,
+        ('ILifeCycle', 'custody_period'): get_custody_period_value,
+        ('ILifeCycle', 'retention_period'): get_retention_period_value,
+        ('IClassification', 'public_trial'): get_public_trial_value,
+        ('IClassification', 'classification'): get_classification_value,
+        ('IClassification', 'privacy_layer'): get_privacy_layer_value,
+    }
+
+
+class Reindexer(object):
+
+    def __init__(self, fixer):
+        self.fixer = fixer
+        self.catalog = fixer.catalog
+        self.stats = fixer.stats
+
+        self.fixer = None
```

Comment:
this line is weird, it sets `fixer` to `None` but it is just assigned above. `self.fixer` does not seem to be read, maybe the two assignments are no longer necessary?